WiseBuy/Managers/APIManager.m

```diff
@@ -47,10 +50,49 @@ + (void)fetchDealsFromEbayAPI:(NSString *)barcode {
         NSLog(@"%@", error);
         dispatch_semaphore_signal(sema);
       } else {
-//            NSHTTPURLResponse *httpResponse = (NSHTTPURLResponse *) response;
             NSError *parseError = nil;
             NSDictionary *responseDictionary = [NSJSONSerialization JSONObjectWithData:data options:0 error:&parseError];
-            NSLog(@"%@",responseDictionary);
+          
+          NSDictionary *finalResponseDic = responseDictionary[@"findItemsByProductResponse"][0][@"searchResult"][0];
+          NSArray *offerLists = finalResponseDic[@"item"];
```

Comment:
You need to be very careful when are accessing properties like this; arrays do not handle accessing elements out of range well. If you access try to element that doesn't exist, it will crash. 

In this code snippet, if `findItemsByProductResponse` returned an empty array, this code would cause the app to crash. For best code reliability, you want to check that the index exists before you go ahead and try to access the value 

It may be easier if you extract the parsing of the responseDictionary into it's own method so you can easily add the checks and then return the offer list